agent/php_error.c

```diff
@@ -497,6 +593,23 @@ void nr_php_error_cb(int type,
     nr_txn_record_error(NRPRG(txn), nr_php_error_get_priority(type), msg,
                         errclass, stack_json);
 
+    /*
+     * Error Fingerprinting Callback
+     */
+    if (is_error_callback_set()) {
+#if ZEND_MODULE_API_NO < ZEND_8_1_X_API_NO
+      file = nr_strdup(error_filename);
+#else
+      file = nr_strdup(ZSTR_VAL(error_filename));
+#endif
+      klass = nr_strdup(errclass);
+
+      nr_php_error_call_error_group_callback(NRPRG(txn), klass, msg, file,
+                                             stack_json);
```

Comment:
Why not pass `errclass` in directly instead of duping it?

Reply:
Initially I wanted to do this, but `errclass` is defined as a `const char*` in this function, which causes the compiler to error out due to discarding of the `const` qualifier for `add_assoc_string`.

See this failed GHA build for the details: https://github.com/newrelic/newrelic-php-agent/actions/runs/5969170113/job/16194501104#step:7:262